v2/tools/generator/internal/config/type_configuration_test.go

```diff
@@ -77,6 +73,18 @@ func TestTypeConfiguration_WhenAzureSecretsBadlyFormed_ReturnsError(t *testing.T
 	g.Expect(err.Error()).To((ContainSubstring(azureGeneratedSecretsTag)))
 }
 
+func TestTypeConfiguration_WhenExportDeprecated_ReturnsError(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	yamlBytes := loadTestData(t)
+
+	var typeConfig TypeConfiguration
+	err := yaml.Unmarshal(yamlBytes, &typeConfig)
+	g.Expect(err).NotTo(Succeed())
+	g.Expect(err.Error()).To(ContainSubstring("$export is deprecated, use $exportAs instead"))
```

Comment:
Prefer using `MatchError()` instead - it automatically fails if `err` is **nil**, and then applies any nested matcher to the message.